### ISSUE TITLE
[1703.01780] Mean teachers are better role models: Weight-averaged consistency targets improve semi-supervised deep learning results

### ISSUE THREAD

**Issue by riron1206** [riron1206/read_paper#19]
## 概要
https://arxiv.org/abs/1703.01780
Mean Teacher: Temporal Ensembling( https://github.com/riron1206/read_paper/issues/18 )を改良した半教師学習
（Temporal Ensemblingはepochごとでしか予測ラベルを更新しないので学習が遅い）
ネットワークの構造が同じStudentモデルとTeacherモデルを用いる
Studentモデルは普通に各stepで重みを更新し、各stepでのStudentモデルの重みの移動平均値をTeacherモデルの重みにする（Studentモデルの重みの平均を使うからMean Teacher）
これによりstepごとのTemporal Ensemblingのような学習ができ、Teacherモデルが最終的なモデルになる

※Teacherモデルの重みの移動平均計算式: θ'(t) = αθ'(t-1) + (1 - α)*θ(t)
θ'(t) :学習stepがtの時のTeacherモデルの重み
α: ハイパーパラメータ
θ(t) :学習stepがtの時のStudentモデルの重み

![model](https://qiita-user-contents.imgix.net/https%3A%2F%2Fuser-images.githubusercontent.com%2F544269%2F35131133-af6bb794-fd07-11e7-84bf-113e363cf980.png?ixlib=rb-1.2.2&auto=format&gif-q=60&q=75&s=4fb9c622d745426100908f068ced963d)

## 新規性・差分
Student-Teacherモデル + consistency cost を導入
consistency costはTeacherモデルとStudentモデルの予測ラベルの差
論文ではMSEを使ってるがKL-divergenceでも良い

## 手法
- StudentとTeacherモデルに同じミニバッチを入れる。ただし、data augmentation（ガウスノイズやランダム変換）は別々に実行
- 以下のようにcost(loss)計算して最適化。正解ラベルあり/なしで式違う
- 正解ラベルありの時
    - classification cost(Softmax Cross Entropy) と consistency costを求める。totalcost=classification_cost+w(t)×consistency_cost
- 正解ラベルなしの時
    - 正解ラベルがなくclassification costを求めることができないため、consistency costのみを用いる。total_cost=w(t)×consistency_cost

## 結果
Mean TeacherとResNetsを組み合わせてImageNetの10%ラベルでSOTA(2017年時点での)

## コメント
- 公式?github: https://github.com/CuriousAI/mean-teacher

<br>

参考サイト
- https://hotcocoastudy.hatenablog.jp/entry/2020/01/23/210357
- https://www.cnblogs.com/wuliytTaotao/p/12825797.html
- https://www.slideshare.net/harmonylab/mean-teacher
- https://qiita.com/H_satomi/items/0836d35d0283e7e8c691
